core/src/test/java/org/elasticsearch/search/aggregations/metrics/cardinality/InternalCardinalityTests.java

```diff
@@ -73,6 +87,41 @@ protected void assertReduced(InternalCardinality reduced, List<InternalCardinali
         }
     }
 
+    public void testFromXContent() throws IOException {
+        InternalCardinality cardinality = createTestInstance();
+        String type = cardinality.getWriteableName();
+        String name = cardinality.getName();
+        ToXContent.Params params = new ToXContent.MapParams(Collections.singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
+        boolean humanReadable = randomBoolean();
+        XContentType xContentType = XContentType.JSON; //randomFrom(XContentType.values());
+        BytesReference originalBytes = toXContent(cardinality, xContentType, params, humanReadable);
+        ParsedCardinality parsed;
+        try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+            assertEquals(type + "#" + name, parser.currentName());
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            parsed = (ParsedCardinality)  parser.namedObject(Aggregation.class, type, name);
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+        assertEquals(cardinality.getName(), parsed.getName());
+        assertEquals(cardinality.getValue(), parsed.getValue(), Double.MIN_VALUE);
```

Comment:
Thats just the expected delta between expeted/actual value for double comparissons in assertEquals()